### DISCUSSION TITLE
Running static app as SPA

### DISCUSSION THREAD

**Discussion by oleksii-matiunin-hb** [vercel/next.js#73273] [App Router]
Hi there,

I'm trying to run Next (14.2.13) as SPA using:
- output: 'export'
- S3 bucket
- NO runtime server

The only solution I could achieve so far is just static website with reloads the page on every redirect. Could someone guide me how  I can move forward and get just one page with loading different bundles?

Thank you

**Comment by denis23x** [comment]
Hi @oleksii-matiunin-hb 

I don't think this is possible or easy to do. Next is just not about that.. even if it is somehow possible, it will probably be problematic and painful in the long run.

If you need SPA use React 🤷‍♂️

To avoid reloads on redirects, use `<Link>` instead `<a>`

**Reply by icyJoseph** [reply]
Next.js uses React. And IMHO, what we Next.js does, is still SPA. It just has multiple potential entry points. 

This contention of comparing Next.js to React, is not helping beginners at all. Perhaps, you mean, use, Waku, Vite's React setup, or Parcel, or whichever.

`export` mode, should just do the trick though. @oleksii-matiunin-hb what issues are you experiencing? `next/link`, ought to be enough. 

If you see the exported folder, you'll find a bunch of `.html` documents, those are potential entry points into your app. Once there, you can SPA transition into other routes, through the Router imperative methods, or `next/link`.
